server/athenian/api/experiments/preloading/entries.py

```diff
@@ -135,11 +143,11 @@ async def load_open_pull_request_facts_unfresh(cls,
         df = cached_df.df
         mask = (
             (df["acc_id"] == account)
-            & df["pr_node_id"].isin(prs)
+            & df["pr_node_id"].isin(pr.encode() for pr in prs)
         )
 
         if exclude_inactive:
-            date_range = build_days_range(time_from, time_to)
+            date_range = set(d.strftime("%Y-%m-%d") for d in build_days_range(time_from, time_to))
             mask &= df["activity_days"].apply(lambda v: bool(set(v).intersection(date_range)))
```

Comment:
This will work very slow. There is vectorized code using `np.searchsorted`, for example https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/features/github/pull_request_metrics.py#L538

Reply:
Anyways, `df["activity_days"]` should not have the string type. It should be an object, a list of datetimes.

Reply:
Indeed, I just realized with the tests failing for PG in CI, it should be like this actually:
```
        if exclude_inactive:
            date_range = build_days_range(time_from, time_to)
            if pdb.url.dialect == "sqlite":
                date_range = set(d.strftime("%Y-%m-%d") for d in date_range)
            mask &= df["activity_days"].apply(lambda v: bool(set(v).intersection(date_range)))
```

I'll have a look at your vectorized implementation.

Reply:
I assume `df["activity_days"]` with SQLite should be coerced to the same format as Postgres inside the preloader.

Reply:
Ah, good point! coercing in the preloader is the best solution, thanks!

Reply:
done